spec/defines/unit_file_spec.rb

```diff
@@ -267,6 +267,16 @@
             expect(subject).not_to create_systemd__daemon_reload(title)
           }
         end
+
+        context 'notify_service => false' do
+          let(:params) { { enable: true, notify_service: false } }
+
+          it do
+            is_expected.not_to contain_service(title).with_subscribes_to(
```

Comment:
`with_subscribes_to` expects a property to be set, like
```puppet
service { $title:
  subscribes_to => File[$file],
}
```

That's never going to be true. This checks the compiled catalog (which mirrors the check on line 210 for example):
```suggestion
            is_expected.not_to contain_service(title).that_subscribes_to(
```

Reply:
That not work, but something like this look ok:
```diff
           let(:params) { { enable: true, notify_service: false } }
 
           it do
-            is_expected.not_to contain_service(title).with_subscribes_to(
-              "File[/etc/systemd/system/#{title}]"
+            is_expected.not_to contain_file("/etc/systemd/system/#{title}").with(
+              notify: /Service\[#{title}\]/
             )
           end
         end
```

It's good for you ?

Reply:
I've try before, with my path when it's true that fail correctly:
```diff
diff --git a/spec/defines/unit_file_spec.rb b/spec/defines/unit_file_spec.rb
index a0521aa..6dde40a 100644
--- a/spec/defines/unit_file_spec.rb
+++ b/spec/defines/unit_file_spec.rb
@@ -272,8 +272,18 @@ describe 'systemd::unit_file' do
           let(:params) { { enable: true, notify_service: false } }
 
           it do
-            is_expected.not_to contain_service(title).with_subscribes_to(
-              "File[/etc/systemd/system/#{title}]"
+            is_expected.not_to contain_file("/etc/systemd/system/#{title}").with(
+              notify: /Service\[#{title}\]/
+            )
+          end
+        end
+
+        context 'notify_service => true' do
+          let(:params) { { enable: true, notify_service: true } }
+
+          it do
+            is_expected.not_to contain_file("/etc/systemd/system/#{title}").with(
+              notify: /Service\[#{title}\]/
             )
           end
         end
```

```
...
  21) systemd::unit_file supported operating systems on centos-8-x86_64 notify_service => true is expected not to contain File[/etc/systemd/system/test.service] with notify =~ /Service\[test.service\]/
      Failure/Error:
        is_expected.not_to contain_file("/etc/systemd/system/#{title}").with(
          notify: /Service\[#{title}\]/
        )
      
        expected that the catalogue would not contain File[/etc/systemd/system/test.service]
      # ./spec/defines/unit_file_spec.rb:285:in `block (6 levels) in <top (required)>'

Finished in 1 minute 49.83 seconds (files took 54.25 seconds to load)
609 examples, 21 failures
...
```

Reply:
Hello,

Any update/idea about this test ?